src/plugins/intel_npu/src/plugin/npuw/partitioning/online/snapshot.cpp

```diff
@@ -386,15 +386,23 @@ void Snapshot::markInternalCompute() {
             prod_cons_tags.insert(group_cons->specialTags());
         }
         if (prod_cons_tags.size() == 1 && !(*prod_cons_tags.begin()).empty()) {
-            NPUW_ASSERT(!group->srcNodes().empty());
-            auto prod_nh = group->srcNodes().at(0);  // all tags are the same, pick either group
-            Group::GPtr group_prod = m_graph->meta(prod_nh).get<Group::GPtr>();
-            NPUW_ASSERT(!group_prod->isolatedTag().empty());
-            if (group_prod->isolatedTag() !=
+            Group::GPtr group_with_tag = nullptr;
+            if (group->srcNodes().empty()) {
+                NPUW_ASSERT(!group->dstNodes().empty());
+                auto cons_nh = group->dstNodes().at(0);  // all tags are the same, pick either group
+                group_with_tag = m_graph->meta(cons_nh).get<Group::GPtr>();
+            } else {
+                auto prod_nh = group->srcNodes().at(0);  // all tags are the same, pick either group
+                group_with_tag = m_graph->meta(prod_nh).get<Group::GPtr>();
+            }
+
+            NPUW_ASSERT(group_with_tag);
+            NPUW_ASSERT(!group_with_tag->isolatedTag().empty());
+            if (group_with_tag->isolatedTag() !=
                 "compute") {  // this pass only operates with "compute" tag set by COMPUTE pipeline
                 continue;
             }
-            group->isolate(group_prod->isolatedTag());
+            group->isolate(group_with_tag->isolatedTag());
```

Comment:
Can you briefly explain the reason to this change?

Reply:
I matched an RMSNorm pattern with this structure:
SomeGroups->RMSNorm->...
So RMSNorm have the isolate tag and in this function we are trying to add an isolate tag for the groups between 2 other isolated group with the same tag.
In the example above there is no source group for the first group, which wasn't the case before (thus assert triggered). I think it still falls into the same category of "in-between" isolated groups.

Reply:
So the previous assumption was that we only mark _internal_ compute, that is, the untagged groups in-between the two tagged ones. So in a layout like this:

```
    A --------> B -------> C -----> D ------> E
  (no tag)  "compute"  (no tag)  "compute"  (no tag)
```

Only C could've been marked as "compute" with the previous algorithm.

With your change, it seems "A" and  "E" should get a "compute" tag too? Wouldn't it draw the whole graph as "compute" ?

I believe, if A would have an extra direct connection to, e.g., E, we wouldn't mark A and E as "compute" in this case there will be two tags in the set?

Reply:
That's right. If the structure above has the exact connections, we would mark A, C and E as "compute". If there is a connection A -> E then we wouldn't mark neither A nor E as "compute". If we have something like
     A -------> B -------> C
(no tag)    (no tag)   "compute"
then we don't mark anything as "compute".
This change won't affect currently supported models as previously we had an assert here. It's only for 1 particular model we enabling now

Reply:
Sounds scary but ok. Please file a task to cover these corner cases with a test.